### DISCUSSION TITLE
Redstone input

### DISCUSSION THREAD

**Discussion by Urbanfirelawz** [cc-tweaked/CC-Tweaked#2039] [Q&A]
trying to figure out the CC for using a redstone input from to fire two different redstone signals for my create world can cant seem to figure it out anyone got any suggestions or help with the coding?  Using Fabric 1.20.1 on my Multiplayer Server

**Comment by Lupus590** [comment]
Have you looked at the docs for the [redstone API](https://tweaked.cc/module/redstone.html) and the [redstone event](https://tweaked.cc/event/redstone.html)?

**Reply by Urbanfirelawz** [reply]
So I have but I don't understand what i have to type in for it when i do the getInput redstone code i get no program.

**Reply by fatboychummy** [reply]
Sounds like you are trying to write code in the shell, but the shell is for commands only. Commands are lua files that you write (or are pre-written) and may take a number of arguments. For example, the command `list` (alias `ls`) allows you to see what files are in the current directory.

You need to run `edit filename.lua` (replacing `filename.lua` with the name of the file you wish to create), then you can start writing Lua code. If you want to rename a program, `mv oldname.lua newname.lua`).

Once you have your program made, you can simply type the name of the program into the shell, and it will run. If you want your program to always be running, set the name of the program to `startup.lua`, and it will be re-ran when the chunk reloads (computers shutdown when the chunk is unloaded).

A small example program with explanations of each line using comments (`--` and the text after `--`) is below. Copy it to your world and mess around with it if you'd like, though you do not need to copy the comments (if you don't want them) as they do not change how the program works, they're just for explaining what does what.
```lua
while true do -- Loop forever
  os.pullEvent("redstone") -- Wait for the computer to detect a redstone change.
  print("Redstone input has changed!") -- Write a message to the screen.

  if rs.getInput("left") and rs.getInput("right") then -- If there is redstone input on the left and right sides of the computer...
    print("Opening door!")
    rs.setOutput("front", true) -- Set the redstone output on the front side to "on" (`true`)
  else -- If one (or both) sides do not have redstone input...
    print("Closing door!")
    rs.setOutput("front", false) -- Set the redstone output on the front side to "off" (`false`)
  end -- Marks the end of the if/else statement.
  
  print("Done checking the redstone change.")
end -- Marks the end of the `while true do` loop. Once we reach here, we jump back to the beginning.
```
The program above will wait for redstone changes, then check if both the left and right side of the computer are receiving redstone inputs (say, from two hidden levers). If it is receiving redstone input from both sides, it'll output redstone from the front side, otherwise it will not output redstone at all. Essentially, it'll act as a redstone AND gate.
